langchain/document_loaders/bilibili.py

```diff
@@ -0,0 +1,83 @@
+import json
+import re
+import warnings
+from typing import List, Tuple
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class BiliBiliLoader(BaseLoader):
+    """Loader that loads bilibili transcripts."""
+
+    def __init__(self, video_urls: List[str]):
+        """Initialize with bilibili url."""
+        self.video_urls = video_urls
+
+    def load(self) -> List[Document]:
+        """Load from bilibili url."""
+        results = []
+        for url in self.video_urls:
+            transcript, video_info = self._get_bilibili_subs_and_info(url)
+            doc = Document(page_content=transcript, metadata=video_info)
+            results.append(doc)
+
+        return results
+
+    def _get_bilibili_subs_and_info(self, url: str) -> Tuple[str, dict]:
+        try:
+            import requests
```

Comment:
The `requests` package is a core dependency of `langchain` so we can import in the outer context (at the top of the file)!

The bilibili_api import looks great, however!